### PR TITLE
Package version sources can be null

### LINKED ISSUE #47: Argument 2 passed to Packagist\Api\Result\Factory::createResult() must be of the type array, null given
When trying to do:

```php
$package = $this->packagistClient->get('sonata-project/classification-media-bundle');
```

I get:

```
PHP Fatal error:  Uncaught TypeError: Argument 2 passed to Packagist\Api\Result\Factory::createResult() must be of the type array, null given, called in /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Result/Factory.php on line 81 and defined in /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Result/Factory.php:103
Stack trace:
#0 /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Result/Factory.php(81): Packagist\Api\Result\Factory->createResult('Packagist\\Api\\R...', NULL)
#1 /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Result/Factory.php(29): Packagist\Api\Result\Factory->createPackageResults(Array)
#2 /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Client.php(219): Packagist\Api\Result\Factory->create(Array)
#3 /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/kn in /home/sullivan/projects/fork/sonata-project/dev-kit/vendor/knplabs/packagist-api/src/Packagist/Api/Result/Factory.php on line 103
```

Only on this package so far.

Ref: https://packagist.org/packages/sonata-project/classification-media-bundle

Using last stable v1.5.0

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Closes #47 

Sometime the `source` field of package versions can be null, we should check this.

Could please make a release after this merge? The library is currently not working at all.

### CODE REVIEW

File: spec/Packagist/Api/Result/FactorySpec.php [hunk 1]
```diff
@@ -79,6 +79,13 @@ function it_creates_packages_with_dependents()
         $this->create($data)->shouldHaveType('Packagist\Api\Result\Package');
     }
 
+    function it_creates_packages_with_null_source()
+    {
+        $data = json_decode(file_get_contents('spec/Packagist/Api/Fixture/get_null_source.json'), true);
```
Comment (greg0ire) [inline spec/Packagist/Api/Result/FactorySpec.php]: otv

File: spec/Packagist/Api/Result/FactorySpec.php [hunk 2]
```diff
@@ -79,6 +79,13 @@ function it_creates_packages_with_dependents()
...
+
+        $this->create($data)->shouldHaveType('Packagist\Api\Result\Package');
```
Comment (greg0ire) [inline spec/Packagist/Api/Result/FactorySpec.php]: You can use the class notation I think

File: spec/Packagist/Api/Result/FactorySpec.php (hunk 1 above)
Comment (Soullivaneuh) [reply spec/Packagist/Api/Result/FactorySpec.php]: Just a copy paste of the other test before this one. I think this class may be re-factored later.

File: spec/Packagist/Api/Result/FactorySpec.php (hunk 2 above)
Comment (Soullivaneuh) [reply spec/Packagist/Api/Result/FactorySpec.php]: Just a copy paste of the other test before this one. I think this class may be re-factored later.

**Comment by NiR-** [comment]
Thank you @Soullivaneuh for this fix :)

**Comment by Soullivaneuh** [comment]
Thank you for the release @NiR-! :+1:

**Pull request merged**